pyvisa_py/usb.py

```diff
@@ -292,6 +295,39 @@ def list_resources() -> List[str]:
                         "usb_interface_number": intfc,
                     },
                 )
+                logging_level = logger.getEffectiveLevel()
+                if logging_level <= logging.DEBUG:
+                    if exc_strs := traceback.format_exception(err):
+                        msg = "Traceback:"
+                        logger.debug(msg)
+                        logger.debug("-" * len(msg))
+                        for exc_str in exc_strs:
+                            for line in exc_str.split("\n"):
+                                logger.debug(line)
+                        logger.debug("-" * len(msg))
+                elif logging_level <= logging.INFO:
+                    if exc_strs := traceback.format_exception_only(err):
+                        logger.info(
+                            "Error raised from underlying module (pyusb): %s",
+                            exc_strs[0].strip(),
+                        )
+
+                dev_path = f"/dev/bus/usb/{dev.bus:03d}/{dev.address:03d}"
```

Comment:
As to your second point, particularly if we are going to implement this for both USB INSTR and USB RAW devices, then it certainly makes sense to move the check logic into a separate module. But I can do that anyway even just to keep the code here cleaner, omitting the gory details that are only pertinent to one OS.

As to your first comment, I thought about gating based on platform. (I also tried to figure out if there's an equivalent for FreeBSD/OS X, but failed to find that.) But I figured by the time I went through all the logic required to check that the `sys.platform` was something this check could work on, I could do it the Python-ic (duck-typing) way and simply look for the file's existence, check the file's permissions, and go from there, and that would essentially take care of "does this check work on my particular platform?" issues.

I'm certainly not married to that way, I get why one might not want to use the duck-type perspective for this check. This almost certainly only works on Linux & nowhere else, and maybe we discover a Linux where it doesn't work, and then we would just want to add to the gating condition. So, yeah, I'll put the check within a `if sys.platform.startswith("linux"): ...` check.

Reply:
> But I can do that anyway even just to keep the code here cleaner, omitting the gory details that are only pertinent to one OS.

Yup, plus it will be easier to test (that's a tricky thing though).

> But I figured by the time I went through all the logic required to check that the sys.platform was something this check could work on, I could do it the Python-ic (duck-typing) way and simply look for the file's existence, check the file's permissions, and go from there, and that would essentially take care of "does this check work on my particular platform?" issues.

This is fair! I prefer an explicit check to make the intent explicit to the readers: "this is only for linux", but my disregard for pythonic way of doing things can and should be questioned :)

> I also tried to figure out if there's an equivalent for FreeBSD/OS X, but failed to find that.

Yeah it's very system-specific, and I don't think we'd have permissions issue on macos (we'd have other ones!). I was wondering if there's a "better" way to get to the device file on linux as well, but it seems that string magic should do the trick until proven otherwise.

Reply:
Here also I think an explicit check will make the intent clearer for people looking at it.